var/spack/repos/builtin/packages/qtgraph/package.py

```diff
@@ -71,9 +72,13 @@ def setup_environment(self, spack_env, run_env):
 
         # The implementor has set up the library and include paths in
         # a non-conventional way.  We reflect that here.
+        if os.path.isdir(self.prefix.lib64):
+            lib_dir = self.prefix.lib64
+        else:
+            lib_dir = self.prefix.lib
```

Comment:
@alalazo Do you know if there is a way to use `self.spec.libs` here? Obviously this dir won't exist at build time, but it will exist at module creation time, correct? We really should separate out `setup_environment` into two separate functions someday.

Reply:
@adamjstewart I'm trying the self.spec.libs suggestion now.   Thanks!   It looks like it is not being used in any packages, when I did a grep for it.

Reply:
It's used in quite a few packages, but it is normally used to look at dependencies, not at the package itself.
```console
$ grep spec.*libs */package.py | wc -l
     232
```

Reply:
If the implementation didn't change in the meanwhile, the key is the subscript operator:
```python
self.spec['qtgraph'].libs
```
should work as expected.

Reply:
@adamjstewart This is what I get:
```
+] /home/jeg/OSS_SPACK/spack/opt/spack/linux-fedora27-x86_64/gcc-7.2.1/qt-5.10.0-qh3dwgmgclmsrdx4al7vmjtqhl56yjb4
==> Installing qtgraph
==> Error: AttributeError: 'Spec' object has no attribute 'libs'

/home/jeg/OSS_SPACK/spack/var/spack/repos/builtin/packages/qtgraph/package.py:82, in setup_environment:
         79
         80        run_env.prepend_path(
         81            'LD_LIBRARY_PATH', join_path(
  >>     82                self.spec.libs, '{0}'.format(self.spec['qt'].version.up_to(3))))
         83
         84        run_env.prepend_path('CPATH', self.prefix.include.QtGraph)
```

Reply:
@jgalarowicz See my comment above.

Reply:
@alalazo @adamjstewart Now get this:
```
==> qt is already installed in /home/jeg/OSS_SPACK/spack/opt/spack/linux-fedora27-x86_64/gcc-7.2.1/qt-5.10.0-qh3dwgmgclmsrdx4al7vmjtqhl56yjb4
==> Installing qtgraph
==> Error: RuntimeError: Unable to recursively locate qtgraph libraries in /home/jeg/OSS_SPACK/spack/opt/spack/linux-fedora27-x86_64/gcc-7.2.1/qtgraph-1.0.0.0-ddoww2gcrpfybd5rg2a3ewe2ct6t56gl

/home/jeg/OSS_SPACK/spack/var/spack/repos/builtin/packages/qtgraph/package.py:82, in setup_environment:
         79
         80        run_env.prepend_path(
         81            'LD_LIBRARY_PATH', join_path(
  >>     82                self.spec['qtgraph'].libs, '{0}'.format(self.spec['qt'].version.up_to(3))))
         83
         84        run_env.prepend_path('CPATH', self.prefix.include.QtGraph)


[jeg@localhost packages]$ 
```

Reply:
@alalazo @adamjstewart I guess qtgraph isn't installed yet.  That path doesn't exist yet.

Reply:
Yeah, the problem is that `setup_environment` is run before build time (to set `spack_env`) and after installation (to set `run_env`). You may have to add a try-except statement. I had to do something like this for all of the intel packages.

Reply:
This works.  It is what you have in the intel packages I looked at.  Thanks for the pointers!
```    68	    def setup_environment(self, spack_env, run_env):
    69	        """Set up the compile and runtime environments for a package."""
    70	        spack_env.set('GRAPHVIZ_ROOT', self.spec['graphviz'].prefix)
    71	        spack_env.set('INSTALL_ROOT', self.prefix)
    72	
    73	        # NOTE: Spack runs setup_environment twice, once pre-build to set up
    74	        # the build environment, and once post-installation to determine
    75	        # the environment variables needed at run-time to add to the module
    76	        # file. The script we need to source is only present post-installation,
    77	        # so check for its existence before sourcing.
    78	        # TODO: At some point we should split setup_environment into
    79	        # setup_build_environment and setup_run_environment to get around
    80	        # this problem.
    81	
    82	        # The implementor has set up the library and include paths in
    83	        # a non-conventional way.  We reflect that here.
    84	        qtgraphlibs = join_path(
    85	            self.spec['qtgraph'].libs,
    86	            '{0}'.format(self.spec['qt'].version.up_to(3)))
    87	
    88	        if os.path.isfile(qtgraphlibs):
    89	            run_env.prepend_path(
    90	                'LD_LIBRARY_PATH', qtgraphlibs)
    91	            run_env.prepend_path('CPATH', self.prefix.include.QtGraph)
```

Reply:
@adamjstewart I haven't pushed the change yet.  I will wait on your advice.

Reply:
@adamjstewart I pushed the change, since I think it will be what you wanted.  If not, I can change the fix.